packages/react/src/components/header/components/headerActionBar/HeaderActionBar.tsx

```diff
@@ -9,9 +9,70 @@ import { useCallbackIfDefined, useEnterOrSpacePressCallback } from '../../../../
 import { HeaderActionBarMenuItem } from '../headerActionBarItem';
 import styles from './HeaderActionBar.module.scss';
 import HeaderActionBarLogo from './HeaderActionBarLogo';
+import { useHeaderContext } from '../../HeaderContext';
 
 const classNames = styleBoundClassNames(styles);
 
+enum TabBarrierPosition {
+  top = 'top',
+  bottom = 'bottom',
+}
+
+type TabBarrierProps = {
+  id: string;
+  tabIndex: number;
+  'aria-hidden': boolean;
+};
+
+const defaultBarrierProps: Partial<TabBarrierProps> = {
+  tabIndex: 0,
+  'aria-hidden': true,
+};
+
+const elementIsInViewport = (element: Element) => {
+  const { top, left, bottom, right } = element.getBoundingClientRect();
+  const { innerHeight, innerWidth } = window;
+
+  return top > 0 && left > 0 && bottom < innerHeight && right < innerWidth;
+};
```

Comment:
should `top>0` be `top>=0` And also in all other comparisons?

I'd also make this a common util function.

Reply:
In my original solution I had them `>=` and for some reason it did not work. Now that I tried again it suddenly worked 🤔 Must have been something else wrong..

Reply:
https://github.com/City-of-Helsinki/helsinki-design-system/pull/1120/commits/6f0eb22ec9872e73f9a22a225882b9717be5c93c